llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -668,6 +668,11 @@ NVPTXTargetLowering::NVPTXTargetLowering(const NVPTXTargetMachine &TM,
     setTruncStoreAction(VT, MVT::i1, Expand);
   }
 
+  setCondCodeAction({ISD::SETNE, ISD::SETEQ, ISD::SETUGE, ISD::SETULE,
+                     ISD::SETUGT, ISD::SETULT, ISD::SETGT, ISD::SETLT,
+                     ISD::SETGE, ISD::SETLE},
+                    MVT::i1, Custom);
```

Comment:
Would `Expand` work here?

Reply:
If it doesn't work out of the box, should we instead expand the implementation of [`TLI.LegalizeSetCCCondCode()`](https://github.com/llvm/llvm-project/blob/dd1c99bac4dc1d5ceeadc79dd31fa12f3e615f18/llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp#L11853-L11983)?

Reply:
The boolean rules I've added are not present in the generic legalizer, so I don't think so. I suppose I could add them there instead, but NVPTX would be the only target using them.

Reply:
If it makes sense for something to live in `.../CodeGen/SelectionDAG/*`, I think we should implement it there instead of in `.../Target/NVPTX/*`. 

This is also my assumption of what @arsenm was asking for [above](https://github.com/llvm/llvm-project/pull/115035#discussion_r1829788004). 

CC @arsenm to correct me.

Reply:
If there is nothing PTX-specific about the expansion (which seems to be the case), I would agree that this could be part of the target-independent expansion. Even if NVPTX is the only backend that uses it, it would be good to make it easier for future backends that may have a use for it.